library/std/src/fs.rs

```diff
@@ -295,6 +295,9 @@ pub fn read_to_string<P: AsRef<Path>>(path: P) -> io::Result<String> {
 /// This function will create a file if it does not exist,
 /// and will entirely replace its contents if it does.
 ///
+/// However, if a parent directory in the path doesn't exist,
+/// the function will fail.
```

Comment:
I think this can be simplified by folding it into the previous paragraph. Something along the lines of

> [...] will create a file if it does not exist - but only if the directory itself exists - and will [...]

Reply:
Indeed, but if we take what @thomcc said into account and introduce hypothetical platforms, this will become hard to fold in the previous paragraph without hindering the readability.